### PR TITLE
Complete review for the Logging reference guide

### PR DESCRIPTION
**Pull request by MichalMaler**
I am proposing a complete review of the Logging guide and its refactoring toward matching its Diataxes reference template. The reasoning behind this action is the desire to bring this content to the RHBQ docs.

I removed the summary and applied the new way of adding `.adoc` metada. Also, I reworded the first sentence, which should be under 26 words, which is now used as the Guide description on the Quarkus docs portal.

After consultations with @Ladicek @mkouba @jmartisk , and @dmlloyd , some content was updated, some content was created and added, and some was moved to a where it fits better.

As part of this initiative, other tech writer reviews will be provided to this content better to approximate the IBM style guide and Diataxes framework.

Includes review and addition of some examples will be added in the follow-up PR.

### CODE REVIEW

**Comment by MichalMaler** [comment]
Currently, adding fixes from an SME review. Will ask you for your opinions soon :)

**Comment by MichalMaler** [comment]
Hello everybody!
I am done with my initial review phase :) @dmlloyd @mkouba @Ladicek . Could I ask you for a review now? :)
Also, thank you so much to @jmartisk who read my latest mvn build and provided me with a quick sanity check, the suggestions of which are already part of this PR.

**Comment by dmlloyd** [comment]
It's a US holiday today but I can look tomorrow.

File: docs/src/main/asciidoc/logging.adoc
Comment (michelle-purcell) [inline docs/src/main/asciidoc/logging.adoc:226]:
```suggestion
Directly:: By adjusting the log level directly.
```

File: docs/src/main/asciidoc/logging.adoc
Comment (michelle-purcell) [inline docs/src/main/asciidoc/logging.adoc:225]:
```suggestion
Globally::  By setting the `quarkus.log.min-level` property.
```

File: docs/src/main/asciidoc/logging.adoc
Comment (michelle-purcell) [inline docs/src/main/asciidoc/logging.adoc:224]: Fix capitalization in list. Would a desc list work here too?


```suggestion
Per category :: As shown in the previous example.
```

File: docs/src/main/asciidoc/logging.adoc
Comment (michelle-purcell) [inline docs/src/main/asciidoc/logging.adoc:231]:
```suggestion
Setting `INFO` as the minimum logging level sets lower-level checks, such as `isTraceEnabled`, to `false`.
```

File: docs/src/main/asciidoc/logging.adoc
Comment (michelle-purcell) [inline docs/src/main/asciidoc/logging.adoc:380]: Does the restructuring and styling in this PR include the included file content? 
Note that this shows up in the preview with some grammatical mistakes.

File: docs/src/main/asciidoc/logging.adoc
Comment (michelle-purcell) [inline docs/src/main/asciidoc/logging.adoc:390]:
```suggestion
Quarkus includes several different log handlers: **console**, **file**, and **syslog**.
```

File: docs/src/main/asciidoc/logging.adoc
Comment (michelle-purcell) [inline docs/src/main/asciidoc/logging.adoc:671]: Should this section (and the other big included file with a similar properties list) be on its own reference page, (as per diataxis) and referenced from a how-to? Ignore if this is the plan for a later iteration. Thanks.

**Review by michelle-purcell — APPROVED** [review]
Great job @MichalMaler :+1: - See my comments. Thanks.

There are some grammatical issues in the included {generated} files too that will need to be fixed at some stage. Just flagging in case these are not on the plan.

File: docs/src/main/asciidoc/logging.adoc
Comment (dmlloyd) [inline docs/src/main/asciidoc/logging.adoc:168]: I think it could be better here to say "No equivalent" or just give a "`-`" like down below.

File: docs/src/main/asciidoc/logging.adoc
Comment (dmlloyd) [inline docs/src/main/asciidoc/logging.adoc:186]: Much of this information is redundant with respect to the table up above. Is there some way to combine this information?

File: docs/src/main/asciidoc/logging.adoc
Comment (dmlloyd) [reply docs/src/main/asciidoc/logging.adoc:222]: I would add something like "... in order for the log messages to be visible in the log".  The user might not always want TRACE messages to be visible by default.

File: docs/src/main/asciidoc/logging.adoc
Comment (dmlloyd) [inline docs/src/main/asciidoc/logging.adoc:228]:
```suggestion
The minimum logging level sets a floor level for which Quarkus needs to generate supporting code, potentially opening the door to optimization opportunities by eliding logging for levels that can never be activated.
```

File: docs/src/main/asciidoc/logging.adoc
Comment (dmlloyd) [reply docs/src/main/asciidoc/logging.adoc:231]: I would use even stronger language: "... causes check methods for lower log levels, such as `isTraceEnabled`, to always return a constant `false` value."

File: docs/src/main/asciidoc/logging.adoc
Comment (dmlloyd) [inline docs/src/main/asciidoc/logging.adoc:259]: The handler configuration is down below so *instead* of doing this here, I think that each handler down below (file, console, etc.) should have an example of global and per-category configuration.

File: docs/src/main/asciidoc/logging.adoc
Comment (MichalMaler) [reply docs/src/main/asciidoc/logging.adoc:186]: Yeah, I could add another table column and put it there. Would that be OK in your books, @dmlloyd ?

Comment (MichalMaler) [reply docs/src/main/asciidoc/logging.adoc:186]: No, this is not the solution. The table maps the values with the numeric value. The lines bellow are descriptions of these levels. Not seeing much duplication here. :)

File: docs/src/main/asciidoc/logging.adoc
Comment (MichalMaler) [reply docs/src/main/asciidoc/logging.adoc]: Thank you, David. Will apply.

File: docs/src/main/asciidoc/logging.adoc
Comment (MichalMaler) [reply docs/src/main/asciidoc/logging.adoc:258]: I agree. Please, if you can provide me these examples, I will add them there. Or would you rather do it in separate PR when this is merged?

File: docs/src/main/asciidoc/logging.adoc
Comment (MichalMaler) [reply docs/src/main/asciidoc/logging.adoc:213]: Nice one. Will apply!

File: docs/src/main/asciidoc/logging.adoc
Comment (dmlloyd) [reply docs/src/main/asciidoc/logging.adoc:186]: The section here is titled "Logging levels" but the first thing that is in the section is a table comparing the standard levels to JUL. At the very least, the table should be *after* the description of what the standard levels are and what they mean, and what the JUL levels are.

File: docs/src/main/asciidoc/logging.adoc
Comment (dmlloyd) [inline docs/src/main/asciidoc/logging.adoc:184]: I'd probably just drop this note altogether; I don't think it answers any question that a user might have.

File: docs/src/main/asciidoc/logging.adoc
Comment (dmlloyd) [inline docs/src/main/asciidoc/logging.adoc:206]:
```suggestion
FINEST:: Increased debug output compared to `TRACE`, which might have a higher frequency.
```

File: docs/src/main/asciidoc/logging.adoc
Comment (mkouba) [inline docs/src/main/asciidoc/logging.adoc:122]: @Ladicek is this still true? We changed the implementation recently if I'm not mistaken...

Comment (Ladicek) [reply docs/src/main/asciidoc/logging.adoc:122]: The change was only for tests. Non-transformed calls to `Log` will still throw during "production" run.

File: docs/src/main/asciidoc/logging.adoc
Comment (mkouba) [reply docs/src/main/asciidoc/logging.adoc:186]: I think that we could just skip the table and keep those two lists ("Log levels used by Quarkus" and `java.util.logging` mapping) because the numerical level value is not very useful... WDYT?

File: docs/src/main/asciidoc/logging.adoc
Comment (mkouba) [reply docs/src/main/asciidoc/logging.adoc:122]: I see.

File: docs/src/main/asciidoc/logging.adoc
Comment (MichalMaler) [reply docs/src/main/asciidoc/logging.adoc:186]: I am for keeping it, but also for the LOG4J and Sl4f note deletion and moving it lover as @dmlloyd suggested.

File: docs/src/main/asciidoc/logging.adoc
Comment (MichalMaler) [inline docs/src/main/asciidoc/logging.adoc:6]: Hello everybody. Thank you so much for your suggestions and review provided. Before we can merge this, I suggest making a small vote on a better name and abstract so that we will have something you like while being ok with some new heading rules we have had in the Quarkus repo for some time now.


Here are my candidates for the heading.
Since this whole guide is a reference, we need to choose a heading like these:

= Configuration guidance for logging
= Logging configuration methods
= Logging configuration parameters
= Logging configuration


We cant use `Configure`, or `How to configure`, because these would be headings for a Tutorial or HowTo template, not reference. We also can use the word `reference` in the title.

For the abstract:

Here is something that should summarize it well, and yes, it also contains some portal title page "fluff". Why is it there..for searchability (google search results), we need to use some Keywords that help us get needed docs on the radar. I am also for a minimalistic approach.

Now, why we can't use something like this:
"Learn how to perform logging in your Quarkus application and configure logging output."
It is because this is not a guide that should teach you anything.

Here are some ideas. We can also create a combination.
1)
Leverage the available information on logging in Quarkus to configure it according to your specific requirements effectively.

2) Use the available information on logging in Quarkus to configure it efficiently according to your specific requirements.

3)
Use the available information on logging in Quarkus and configuring the logging output accordingly.

4)
Configure logging in your Quarkus applications to determine how data and messages about events are recorded and made available. Use the available information to set the appropriate configuration properties that you need.

Can I have your 2cents on this too?
@dmlloyd @Ladicek @mkouba @michelle-purcell

File: docs/src/main/asciidoc/logging.adoc
Comment (MichalMaler) [reply docs/src/main/asciidoc/logging.adoc:379]: Hello @michelle-purcell . Yes, I will do that in a separate PR. My goal is to merge this asap as we get some conversation about the heading and abstract.
Before that happens, I will start patching this included files from a new branch.

File: docs/src/main/asciidoc/logging.adoc
Comment (Ladicek) [reply docs/src/main/asciidoc/logging.adoc:6]: The guide explains:

1. using logging API(s)
2. logger configuration

I'd just call it "Logging" and be done with it. It is the ultimate logging reference guide after all.

When it comes to the summary, I'll try to distill the absolute gist of it:

> Use the logging API in Quarkus efficiently. Configure logging output according to your needs. Use logging adapters to unify the output from other logging APIs.

I think those are the main topics covered in this reference.

I passionately dislike padding words like "Use available information" (or even "utilize" or "leverage"!), so I'm trying to avoid them myself :-)

File: docs/src/main/asciidoc/logging.adoc
Comment (MichalMaler) [reply docs/src/main/asciidoc/logging.adoc:257]: Great, I will star a PR for a minimalistic review of Logging guide includes. I think that will be great opportunity to add these examples.

File: docs/src/main/asciidoc/logging.adoc
Comment (Ladicek) [inline docs/src/main/asciidoc/logging.adoc:223]: Not sure what this means. This doesn't follow from the previous text. Something like this should be enough:

```asciidoc
When you set the log level to below `DEBUG`, you must also adjust the minimum log level.
This setting is either global, using the `quarkus.log.min-level` configuration property, or per category:

[source, properties]
----
quarkus.log.category."org.hibernate".min-level=TRACE
----

It sets a floor level for which Quarkus needs to generate supporting code.
The minimum log level must be set at build time, so that Quarkus can open the door to optimization opportunities by eliding logging for levels that can never be used.
```

File: docs/src/main/asciidoc/logging.adoc
Comment (Ladicek) [inline docs/src/main/asciidoc/logging.adoc:235]:
```suggestion
If you add these properties on the command line, ensure the `"` character is escaped properly:
```

File: docs/src/main/asciidoc/logging.adoc
Comment (Ladicek) [inline docs/src/main/asciidoc/logging.adoc:255]:
```suggestion
If you want to configure something extra for a specific category, create a named handler like `quarkus.log.handler.[console|file|syslog].<your-handler-name>.*` and set it up for that category by using `quarkus.log.category.<my-category>.handlers`.
```

File: docs/src/main/asciidoc/logging.adoc
Comment (Ladicek) [inline docs/src/main/asciidoc/logging.adoc:290]:
```suggestion
* The parent category is examined if no level configuration exists for a given logger category.
```

File: docs/src/main/asciidoc/logging.adoc
Comment (Ladicek) [inline docs/src/main/asciidoc/logging.adoc:462]: This note probably belongs a bit earlier, it is not specific to the Syslog handler.

File: docs/src/main/asciidoc/logging.adoc
Comment (Ladicek) [inline docs/src/main/asciidoc/logging.adoc:558]: `build.gradle` is not XML.

```suggestion
For Gradle, add the following configuration to the `build.gradle` file:
```

File: docs/src/main/asciidoc/logging.adoc
Comment (MichalMaler) [reply docs/src/main/asciidoc/logging.adoc:223]: Deluxe, applying :)

File: docs/src/main/asciidoc/logging.adoc
Comment (MichalMaler) [reply docs/src/main/asciidoc/logging.adoc:253]: Good catch. Adding.

File: docs/src/main/asciidoc/logging.adoc
Comment (MichalMaler) [reply docs/src/main/asciidoc/logging.adoc:462]: Nice one. Moving it to the `=== Root logger configuration` chapter.
Thank you!

File: docs/src/main/asciidoc/logging.adoc
Comment (MichalMaler) [reply docs/src/main/asciidoc/logging.adoc:257]: Yeah, the link was pointing to an unrelated chapter, so my description was wrong.
Thank you for spotting this.

```
Thus, configurations made under `quarkus.log.console.*`, `quarkus.log.file.*`, and `quarkus.log.syslog.*`, are global and apply for all categories. For more information, see <<loggingConfigurationReference>>.
```

**Review by Ladicek — APPROVED** [review]
LGTM

**Comment by Ladicek** [comment]
This looks good from my perspective.

Not sure if any other review is needed, I'm relatively new to the docs process.

**Comment by MichalMaler** [comment]
> This looks good from my perspective.
> 
> Not sure if any other review is needed, I'm relatively new to the docs process.

Thank you so much for your time dedicated to this PR! You suggestions really took this to another level.
Regarding the process, most of time, we are good after two SME checks and one writer's peer approval :)

**Comment by MichalMaler** [comment]
@jmartisk Hello Jan! Thx for the additional review. Applied your suggestions on the line 351:

```
The `quarkus-logging-json` extension may be employed to add support for the JSON logging format and its related configuration.
```

Please, if you dont see anything else, please approve and feel free to merge if pipelines succeed.

Cheers

**Review by jmartisk — APPROVED** [review]

**Comment by MichalMaler** [comment]
@gsmet Hello Guillaume, hope you are doing well. Please, I added a back-port label to this review. I review was big, but the name of the files are the same. I think it should be tagged some time ago to get to the latest release, but I was on PTO and somehow it slipped trough. Will you add this to the next batch or maybe even add to the current one and republish if that is not much time consuming?
Thank you.

**Comment by gsmet** [comment]
Please don't add back the backport label when I already backported a PR. See the history above.

**Pull request merged**
